# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.cpp
# Commit: 2865fe682139
# Full Hash: 2865fe682139450ad7574810366f980d771412c2
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2023-09-12 04:12:49
# Regressor Bug: 1690111
# File Overlap Count: 1
# Description:
#   Bug 1690111 - Implement the ability to pin lengths (prevent changes to length, eg via detaching) for ArrayBuffers and their views. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D131301
# ==============================================================================

diff -r 9dcd2416f8a5 -r 2865fe682139 js/src/jit/MacroAssembler.cpp
--- a/js/src/jit/MacroAssembler.cpp	Mon Sep 11 12:52:18 2023 +0000
+++ b/js/src/jit/MacroAssembler.cpp	Mon Sep 11 12:52:19 2023 +0000
@@ -6821,8 +6821,8 @@
                Address(temp, ObjectElements::offsetOfFlags()),
                Imm32(ObjectElements::SHARED_MEMORY), &done);
 
-  // An ArrayBufferView with a null buffer has never had its buffer exposed to
-  // become detached.
+  // An ArrayBufferView with a null/true buffer has never had its buffer
+  // exposed, so nothing can possibly detach it.
   fallibleUnboxObject(Address(obj, ArrayBufferViewObject::bufferOffset()), temp,
                       &done);
 